Azure/communication-ui-library-android
# [Bug] crash on early end

## Purpose

* Fix for crash on early back. Instead of not handling the exception (endCall called before start), we pass the error down, so it can bubble to the error handler. It now results in a prompt instead of a crash.

We might want to follow up and squash this error as well, but crash is the priority for this PR. We could return a CompletableFuture that does nothing instead of passing the error through, or we can follow up and ensure endCall() isn't called before startCall() so the condition for the error doesn't exist.

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[X] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* Test the code

```
Go to setup screen, and back. Do not join call. App should not crash.
```

## What to Check
Verify that the following are valid
* ...

## Other Information
